annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+
+    assert doc_objs["doc1"]["objects"][0]["id"] == 101
+    assert doc_objs["doc1"]["objects"][0]["links"] == [101]
+    assert doc_objs["doc1"]["objects"][0]["children"] == [3]
+    assert doc_objs["doc1"]["objects"][1]["id"] == 102
+    assert doc_objs["doc1"]["objects"][2]["id"] == 103
+
+    mock_get_new_id.assert_called()
+    mock_get_common_ids.assert_called()
+    mock_get_links_and_children.assert_called()
+    mock_get_common_values.assert_called()
+
+
+def test_change_ids_with_children():
+    all_tasks_objs = {
+        "task1": {"objects": [{"id": 1}, {"id": 2}, {"id": 3}]},
+        "task2": {"objects": [{"id": 4}, {"id": 5}, {"id": 6}]},
+    }
+
+    doc_objs = {
+        "doc1": {
+            "objects": [
+                {"id": 1, "links": [2], "children": [3, 4]},
+                {"id": 2},
+                {"id": 3},
+            ]
+        }
+    }
+
+    id_mapping = {(1,): 101, (2,): 102, (3,): 103, (4,): 104}
+
+    mock_get_new_id = MagicMock(
+        side_effect=lambda old_id, _: id_mapping.get((old_id,), old_id)
+    )
+    mock_get_common_ids = MagicMock(
+        side_effect=lambda old_id, _: [id_mapping.get((old_id,), old_id)]
+    )
+    mock_get_links_and_children = MagicMock(
+        side_effect=lambda ids, _, __: (
+            [id_mapping.get((i,), i) for i in ids],
+            [id_mapping.get((i,), i) for i in ids],
+        )
+    )
+    mock_get_common_values = MagicMock(side_effect=lambda objs: objs)
+
+    with patch("annotation.tasks.services.get_new_id", mock_get_new_id), patch(
+        "annotation.tasks.services.get_common_ids", mock_get_common_ids
+    ), patch(
+        "annotation.tasks.services.get_links_and_children",
+        mock_get_links_and_children,
+    ), patch(
+        "annotation.tasks.services.get_common_values", mock_get_common_values
+    ):
+
+        change_ids(all_tasks_objs, doc_objs, id_mapping)
+
+    assert doc_objs["doc1"]["objects"][0]["id"] == 101
+    assert doc_objs["doc1"]["objects"][0]["links"] == [101]
+    assert doc_objs["doc1"]["objects"][0]["children"] == [101]
+    assert doc_objs["doc1"]["objects"][1]["id"] == 102
+    assert doc_objs["doc1"]["objects"][2]["id"] == 103
+
+    mock_get_new_id.assert_called()
+    mock_get_common_ids.assert_called()
+    mock_get_links_and_children.assert_called()
+    mock_get_common_values.assert_called()
+
+
+def test_remove_ids():
+    task_obj = {
+        "id": 1,
+        "name": "Task 1",
+        "description": "This is a task.",
+        "links": [2, 3],
+        "children": [4, 5],
+        "extra_info": "Some extra info",
+    }
+
+    expected_result = {
+        "name": "Task 1",
+        "description": "This is a task.",
+        "extra_info": "Some extra info",
+    }
+
+    result = remove_ids(task_obj)
+
+    assert result == expected_result
+    assert "id" not in result
+    assert "links" not in result
+    assert "children" not in result
+
+
+@pytest.fixture
+def setup_data():
```

Comment:
Fixture should be at the start of the file

Reply:
We should think about the need for this fixtures, since it is being used in just one test. Consider using it as a local var